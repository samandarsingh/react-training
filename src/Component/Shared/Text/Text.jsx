import React from "react";
import PropTypes from "prop-types";
/**
 * 
 * @param {string} title
 * @return Element 
 */
const Text = props => { 
    const {title}  = props;
    return <span>{title}</span>;    
}
Text.propTypes = {    
    title:  PropTypes.string.isRequired
}
export default Text;