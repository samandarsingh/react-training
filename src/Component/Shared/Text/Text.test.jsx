import React from "react";
import { mount } from "enzyme";
import renderer from 'react-test-renderer';
import Text from "./Text";

describe(`my component should be run`,() => {
    const props ={
        title: ""
    }
    it(`component should be run currectlly`,() => {
        const wrapper = mount(<Text {...props}/>);
        expect(wrapper).ToBeTruthy;
    });
    it(`renders correctly`, () => {    
        const tree = renderer
          .create(<Text {...props}/>)
          .toJSON();
        expect(tree).toMatchSnapshot();
    });
});