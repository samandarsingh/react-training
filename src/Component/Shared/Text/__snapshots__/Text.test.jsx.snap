// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`my component should be run renders correctly 1`] = `
<span>
  
</span>
`;
